### PR TITLE
Add filter for broker telemetry event fields

### PR DESCRIPTION
**Pull request by gitaumoses4**
## Why 
Since we need to do the privacy review for telemetry, for now, we should add a filter to NOT upload anything for which we aren’t sure whether that is PII or OII. 

## How
Add a filter that only allows specific fields to be published to a telemetry relay client

## Broker PR
Here is the related broker PR: https://github.com/AzureAD/ad-accounts-for-android/pull/1931

### CODE REVIEW

File: changelog.txt [hunk 1]
```diff
@@ -13,6 +13,7 @@ Version 6.0.0
 - [PATCH] Fix APPLICATION_CANCELLED by handling back button press (#1725)
 - [PATCH] Added Base64 encode for the AuthorizationRequest state parameter (#1750)
 - [PATCH] Fix missing authority_url when creating the authority audience (#1753)
+- [PATCH] Add filter for broker telemetry event fields.
```
Comment (gitaumoses4) [inline changelog.txt]:
```suggestion
- [PATCH] Add filter for broker telemetry event fields. (#1793)
```

File: common4j/src/main/com/microsoft/identity/common/java/telemetry/relay/AbstractTelemetryRelayClient.java [hunk 2]
```diff
@@ -39,33 +36,33 @@
...
+                relayEvent(filteredData);
+            } catch (TelemetryRelayException e) {
+                Logger.error(methodTag, "Error relaying telemetry data", e);
+            }
         }
     }
 
     /**
-     * Add an event filter to the list of filters
+     * Add an event filter
```
Comment (shahzaibj) [inline common4j/src/main/com/microsoft/identity/common/java/telemetry/relay/AbstractTelemetryRelayClient.java]:
```suggestion
     * Add an event filter.
```

File: common4j/src/main/com/microsoft/identity/common/java/telemetry/relay/AbstractTelemetryRelayClient.java [hunk 3]
```diff
@@ -39,33 +36,33 @@
 public abstract class AbstractTelemetryRelayClient<T> implements ITelemetryObserver<T> {
 
     private static final String TAG = AbstractTelemetryRelayClient.class.getSimpleName();
-    // Filters to apply on the events captured.
-    // This basically performs an AND operation on the filters to determine whether the event will be relayed.
-    private final List<ITelemetryEventFilter<T>> eventFilters = new ArrayList<>();
+    // Filter to apply on the events captured.
+    private ITelemetryEventFilter<T> mEventFilter = null;
```
Comment (shahzaibj) [inline common4j/src/main/com/microsoft/identity/common/java/telemetry/relay/AbstractTelemetryRelayClient.java:40]: It seems we've changed the model of the code here now and only going to allow one filter. What was the rationale behind this change. Are there pros and cons to this approach over the previous approach?

File: common4j/src/main/com/microsoft/identity/common/java/telemetry/relay/AbstractTelemetryRelayClient.java [hunk 4]
```diff
@@ -39,33 +36,33 @@
...
 
-        try {
-            relayEvent(telemetryData);
-        } catch (TelemetryRelayException e) {
-            Logger.error(methodTag, "Error relaying telemetry data", e);
+        if (filteredData != null) {
+            try {
+                relayEvent(filteredData);
+            } catch (TelemetryRelayException e) {
+                Logger.error(methodTag, "Error relaying telemetry data", e);
```
Comment (shahzaibj) [inline common4j/src/main/com/microsoft/identity/common/java/telemetry/relay/AbstractTelemetryRelayClient.java:56]: What might cause a relay exception?

Comment (shahzaibj) [reply common4j/src/main/com/microsoft/identity/common/java/telemetry/relay/AbstractTelemetryRelayClient.java:56]: I'm thinking in this case we may want to hold onto the data in some sort of telemetry cache and make another attempt at relaying it in the future....just something to think about...we don't have to solve that problem right now

File: common4j/src/main/com/microsoft/identity/common/java/telemetry/relay/ITelemetryEventFilter.java [hunk 5]
```diff
@@ -31,7 +31,7 @@
      * Invoked when a new event is captured by the telemetry.
      * @param telemetryEvent the telemetry event data
      *
-     * @return a boolean representing whether the event should be relayed.
+     * @return an event with filtered fields. Return null if the event is to be ignored completely.
      */
-    boolean shouldRelay(T telemetryEvent);
+    T apply(T telemetryEvent);
```
Comment (shahzaibj) [inline common4j/src/main/com/microsoft/identity/common/java/telemetry/relay/ITelemetryEventFilter.java]: Seems like this is `Nullable` -probably want to mark it as such here

**Review by shahzaibj — APPROVED** [review]
Some basic comments...but I wouldn't block on them

**Review by mohitc1 — APPROVED** [review]

File: common4j/src/main/com/microsoft/identity/common/java/telemetry/relay/AbstractTelemetryRelayClient.java (hunk 4 above)
Comment (gitaumoses4) [reply common4j/src/main/com/microsoft/identity/common/java/telemetry/relay/AbstractTelemetryRelayClient.java:56]: > What might cause a relay exception?

Possibly a deleted aria instance.

**Pull request merged**